src/NSwag.SwaggerGeneration.AspNetCore/AspNetCoreToSwaggerGeneratorCommandEntryPoint.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using NSwag.SwaggerGeneration.Processors;
+
+namespace NSwag.SwaggerGeneration.AspNetCore
+{
+    /// <summary>An entry point for the aspnetcore2swagger command.</summary>
+    internal class AspNetCoreToSwaggerGeneratorCommandEntryPoint
+    {
+        public static void Process(string settingsData)
+        {
+            var settings = CreateSettings(settingsData);
+            var serviceProvider = GetServiceProvider(settings.ApplicationName);
+            var apiDescriptionProvider = serviceProvider.GetRequiredService<IApiDescriptionGroupCollectionProvider>();
+
+            var swaggerGenerator = new AspNetCoreToSwaggerGenerator(new AspNetCoreToSwaggerGeneratorSettings());
+            var swaggerDocument = swaggerGenerator.GenerateAsync(apiDescriptionProvider.ApiDescriptionGroups).GetAwaiter().GetResult();
+
+            var outputPathDirectory = Path.GetDirectoryName(settings.Output);
+            Directory.CreateDirectory(outputPathDirectory);
+            File.WriteAllText(settings.Output, swaggerDocument.ToJson());
+        }
+
+        private static IServiceProvider GetServiceProvider(string applicationName)
+        {
+            var assemblyName = new AssemblyName(applicationName);
+            var assembly = Assembly.Load(assemblyName);
+
+            if (assembly.EntryPoint == null)
+            {
+                throw new InvalidOperationException($"Unable to locate the program entry point for {assemblyName}.");
+            }
+
+            var entryPointType = assembly.EntryPoint.DeclaringType;
+            var buildWebHostMethod = entryPointType.GetMethod("BuildWebHost");
```

Comment:
This would get removed once the HostingStartup.Sources package becomes available.